### PR TITLE
Chore/build warnings

### PR DESCRIPTION
**Pull request by solarisstar**

### CODE REVIEW

File: code/game/list.c [hunk 1]
```diff
@@ -58,7 +58,7 @@ static int list_add_impl(list_p list, void* data, dataType_t type, size_t size,
 
     node->cont = (container_p)malloc(sizeof(container));
     if (node->cont == NULL) {
-        return 0;
+        goto on_error;
```
Comment (MJLHThomassen) [inline code/game/list.c:61]: goto? eh....

Comment (solarisstar) [inline code/game/list.c:61]: Why OMFG?

Comment (MJLHThomassen) [inline code/game/list.c:61]: Never mind, it had to do with the goto, see other PR :P

**Comment by MJLHThomassen** [comment]
Duplicated in #31

**Pull request closed without merging**